File: python/sdk/claim_vidaId.py
from pwrpy.pwrwallet import Wallet
from dotenv import load_dotenv
import os
load_dotenv()

# Setting up your wallet in the SDK
seed_phrase = os.getenv("SEED_PHRASE")
wallet = Wallet.new(seed_phrase)

def claim():
    # Add a unique VM ID
    vida_id = 102030

    # Claim the VM ID
    res = wallet.claim_vida_id(vida_id)

    # Error handling
    if res.success:
        print("Transaction Hash:", res.hash.hex())
    else:
        print("Error:", res.message)

if __name__ == "__main__":
    claim()

File: go/sdk/claimVidaId.go
package sdk

import (
	"fmt"
	"os"

	"github.com/joho/godotenv"
	"github.com/pwrlabs/pwrgo/wallet"
)

func ClaimVidaId() {
	godotenv.Load()
	seedPhrase := os.Getenv("SEED_PHRASE")

	// Setting up your wallet in the SDK
	wallet, err := wallet.New(seedPhrase)
	if err != nil {
		fmt.Println("Error:", err)
		return
	}

	// Add a unique VM ID
	vidaId := 102030
	feePerByte := wallet.GetRpc().GetFeePerByte()

	// Claim the VM ID
	tx := wallet.ClaimVidaId(vidaId, feePerByte)

	// Error handling
	if tx.Success {
		fmt.Printf("Transaction Hash: %s\n", tx.Hash)
	} else {
		fmt.Println("Error:", tx.Error)
	}
}
